src/Nox.Types.EntityFramework/Configurations/NoxDatabaseConfigurator.cs

```diff
@@ -65,11 +65,21 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
         {
             foreach (var relationshipToCreate in relationshipsToCreate)
             {
-                // One to ?? (// Many to Many are setup by EF)
-                if (relationshipToCreate.Relationship.ShouldGenerateForeignKeyOnThisSide() && relationshipToCreate.Relationship.WithSingleEntity())
+                // Many to Many
+                // Currently, configured bi-directionally, shouldn't cause any issues.
+                if (relationshipToCreate.Relationship.WithMultiEntity &&
+                    relationshipToCreate.Relationship.Related.EntityRelationship.WithMultiEntity)
+                {
+                    builder
+                        .HasMany(relationshipToCreate.Relationship.Name)
+                        .WithMany(relationshipToCreate.Relationship.Related.EntityRelationship.Name);
+                }
+                // OneToOne and OneToMany, setup should be done only on foreign key side
+                else if (relationshipToCreate.Relationship.ShouldGenerateForeignKeyOnThisSide() &&
+                    relationshipToCreate.Relationship.WithSingleEntity())
                 {
                     //One to Many
-                    if (relationshipToCreate.Relationship.IsManyRelationshipOnOtherSide())
+                    if (relationshipToCreate.Relationship.Related.EntityRelationship.WithMultiEntity)
```

Comment:
Can we reduce property chain invocation? Ideally, it shouldn't invoke more than two nested properties.